tests/pact/tc.repl

```diff
@@ -329,6 +329,16 @@
     "test anon lambdas"
     (map (lambda (i) (> i 1)) [1 2 3]))
 
+  (defpact fail-steps-type-missmatch: integer ()
```

Comment:
need to also catch the problems with yield/resume. this example should typecheck but doesn't:

```
(defpact tc-nesteepact:integer (a:string)
    (step (let ((r 1) (y:object{nestedschema} { 'c: 10.0 })) (yield y) r))
    (step (resume { "c" := c } (+ (length a) 1))))

  (defpact tc-nestedpact:integer (a:string)
    (step (tc-nesteepact a))
    (step (continue (tc-nesteepact a))))
```